### DISCUSSION TITLE
Geometric transformations

### DISCUSSION THREAD

**Discussion by pabloduque0** [deepmind/dm_pix#37] [Ideas]
Hello there! I love this line of work so thank you for open sourcing this one!

I have a few question to understand a bit more the scope of the library in order to asses if its a good fit for what I am attempting.

- Is it on the roadmap to add more processing functions? I am specifically interested on geometric transformations (shear, elastic transformation, piecewise affine, ...). 
- Would it be interesting to have some of the random transformations more flexible, eg. add probability param to `random_flip_up_down` and `random_flip_left_right` as opposed to having a coin flip. 

Happy to contribute to some of the above if they are indeed interesting to this project/library.

Best

**Comment by claudiofantacci** [comment]
Hi @pabloduque0, thanks a lot for your kind words!

Let me answer your question in order:
 - we have no specific roadmap at the moment; this library is mostly community driven: we usually add new functions when provided  by users, or through direct contributions. If you have any function you'd like to see in the library, feel free to send a PR.
 - Definitely interested! I think if we keep the default a coin flip and we add a keyword only param to make it non-coin-flip (biased coin-flip), that would work nicely!

Feel free to send PRs and I'll be more than happy to review and provide suggestions! 🚀
